src/main/java/umk/mat/pajda/ProjektZespolowy/configs/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,61 @@
+package umk.mat.pajda.ProjektZespolowy.configs;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+import umk.mat.pajda.ProjektZespolowy.services.AuthService;
+import umk.mat.pajda.ProjektZespolowy.services.JWTService;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+  private final JWTService jwtService;
+  private final AuthService authService;
+
+  @Override
+  protected void doFilterInternal(
+      HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+      throws ServletException, IOException {
+    final String authHeader = request.getHeader("Authorization");
+    final String jwt;
+    final String userEmail;
+
+    if (StringUtils.isEmpty(authHeader)
+        || !org.apache.commons.lang3.StringUtils.startsWith(authHeader, "Bearer ")) {
+      filterChain.doFilter(request, response);
+      return;
+    }
+    jwt = authHeader.substring(7);
```

Comment:
what is the magic 7 here?

Reply:
if authHeader starts with "Bearer " we need to extract only JWT token and "Bearer " has 7 length, i will add variable to make it more clear

Reply:
i added variable